mm2src/mm2_core/src/mm_ctx.rs

```diff
@@ -346,53 +354,40 @@ impl MmCtx {
 
     pub fn mm_version(&self) -> &str { &self.mm_version }
 
+    /// Retrieves an optional shared connection from the pool for the specified database ID.
+    /// Returns `None` if the connection pool is not initialized.
     #[cfg(not(target_arch = "wasm32"))]
-    pub fn init_sqlite_connection(&self) -> Result<(), String> {
-        let sqlite_file_path = self.dbdir().join("MM2.db");
-        log_sqlite_file_open_attempt(&sqlite_file_path);
-        let connection = try_s!(Connection::open(sqlite_file_path));
-        try_s!(self.sqlite_connection.pin(Arc::new(Mutex::new(connection))));
-        Ok(())
-    }
-
-    #[cfg(not(target_arch = "wasm32"))]
-    pub fn init_shared_sqlite_conn(&self) -> Result<(), String> {
-        let sqlite_file_path = self.shared_dbdir().join("MM2-shared.db");
-        log_sqlite_file_open_attempt(&sqlite_file_path);
-        let connection = try_s!(Connection::open(sqlite_file_path));
-        try_s!(self.shared_sqlite_conn.pin(Arc::new(Mutex::new(connection))));
-        Ok(())
-    }
-
-    #[cfg(not(target_arch = "wasm32"))]
-    pub async fn init_async_sqlite_connection(&self) -> Result<(), String> {
-        let sqlite_file_path = self.dbdir().join("KOMODEFI.db");
-        log_sqlite_file_open_attempt(&sqlite_file_path);
-        let async_conn = try_s!(AsyncConnection::open(sqlite_file_path).await);
-        try_s!(self.async_sqlite_connection.pin(Arc::new(AsyncMutex::new(async_conn))));
-        Ok(())
+    pub fn shared_sqlite_conn_opt(&self, db_id: Option<&str>) -> Option<Arc<Mutex<Connection>>> {
```

Comment:
As I can see, shared_sqlite_conn_opt fn currently is always called with db_id as None. Is there a use case when it is not None?
Looks like db_id is derived from an account, so it seems to me when we use a shared conn we would not want to use db_id derived from multiple accounts

Reply:
https://github.com/KomodoPlatform/komodo-defi-framework/pull/2093#discussion_r1656771659 there hasn't been a use case for where we'd call `shared_sqlite_conn_opt` yet with a `db_id`.